cmd/metal-api/internal/datastore/network.go

```diff
@@ -91,42 +92,48 @@ func (p *NetworkSearchQuery) generateTerm(rs *RethinkStore) *r.Term {
 	}
 
 	for _, prefix := range p.Prefixes {
-		ip, length := utils.SplitCIDR(prefix)
+		pfx, err := netip.ParsePrefix(prefix)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse prefix %w", err)
```

Comment:
It'd be nice if we could validate this in the service layer in order to properly return `BadRequest` status code.

Reply:
BadRequest is not return from the service layer